scripts/generateTranslations.ts

```diff
@@ -589,55 +571,333 @@ class TranslationGenerator {
+            }
 
-                        // Build the translated template expression, referencing the translated template spans as necessary
-                        return this.stringToTemplateExpression(translatedTemplate, translatedComplexExpressions);
+            // Handle the case where the same path has both additions and removals (treat as modified, not deleted)
+            // Also check if removed paths still exist in en.ts (partial removal within function)
+            for (const removedPath of this.pathsToRemove) {
+                if (this.pathsToModify.has(removedPath)) {
+                    this.pathsToRemove.delete(removedPath); // It's modified, not removed
+                } else if (get(en, removedPath) !== undefined) {
+                    // Path still exists in en.ts, so it's modified not removed
+                    this.pathsToRemove.delete(removedPath);
+                    this.pathsToModify.add(removedPath);
+                }
+            }
+
+            // Classify pathsToModify into actual modify vs add based on target file existence
+            // We need to check against each target language file to properly classify paths
+            for (const targetLanguage of this.targetLanguages) {
+                const targetPath = path.join(this.languagesDir, `${targetLanguage}.ts`);
+
+                if (fs.existsSync(targetPath)) {
+                    const existingContent = fs.readFileSync(targetPath, 'utf8');
+                    const existingSourceFile = ts.createSourceFile(targetPath, existingContent, ts.ScriptTarget.Latest, true);
+
+                    // Check each path in pathsToModify to see if it actually exists in this target file
+                    const existingTranslationsNode = this.findTranslationsNode(existingSourceFile);
+                    for (const pathToCheck of this.pathsToModify) {
+                        if (!TSCompilerUtils.objectHas(existingTranslationsNode, pathToCheck)) {
+                            this.pathsToModify.delete(pathToCheck);
+                            this.pathsToAdd.add(pathToCheck);
+                        }
                     }
+
+                    // Break after first existing target file since path classification should be consistent
+                    break;
+                }
+            }
+
+            if (this.verbose) {
+                console.log(`🔄 Paths to modify: ${Array.from(this.pathsToModify).join(', ')}`);
+                console.log(`➕ Paths to add: ${Array.from(this.pathsToAdd).join(', ')}`);
+                console.log(`🗑️ Paths to remove: ${Array.from(this.pathsToRemove).join(', ')}`);
+            }
+        } catch (error) {
+            if (this.verbose) {
+                console.warn('⚠️ Error building paths from git diff, falling back to full translation:', error);
+            }
+            // On error, don't populate any paths (will fall back to full translation)
```

Comment:
This is already called within a `isIncremental` check. I'd think we'd just abort in this case and not do a full translation.